third_party/terraform/tests/data_source_monitoring_notification_channel_test.go

```diff
@@ -0,0 +1,192 @@
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleMonitoringNotificationChannel_byType(acctest.RandomWithPrefix("tf-test")),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceState(
+						"data.google_monitoring_notification_channel.my",
+						"google_monitoring_notification_channel.my"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceGoogleMonitoringNotificationChannel_byDisplayNameAndType(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleMonitoringNotificationChannel_byDisplayNameAndType(acctest.RandomWithPrefix("tf-test")),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceState(
+						"data.google_monitoring_notification_channel.my",
+						"google_monitoring_notification_channel.myemail"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceGoogleMonitoringNotificationChannel_NotFound(t *testing.T) {
+	displayName := acctest.RandomWithPrefix("tf-test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceGoogleMonitoringNotificationChannel_NotFound(displayName),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("No NotificationChannel found using filter=display_name=\"%s\"", displayName)),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceGoogleMonitoringNotificationChannel_NotUnique(t *testing.T) {
+	displayName := acctest.RandomWithPrefix("tf-test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		Providers:                 testAccProviders,
+		PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleMonitoringNotificationChannel_NotUnique(displayName),
+			},
+			{
+				Config:      testAccDataSourceGoogleMonitoringNotificationChannel_NotUniqueDS(displayName),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("More than one matching NotificationChannel found using filter=display_name=\"%s\"", displayName)),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGoogleMonitoringNotificationChannel_byDisplayName(displayName string) string {
+	return fmt.Sprintf(`
+resource "google_monitoring_notification_channel" "my" {
+  display_name = "%s"
+  type         = "webhook_tokenauth"
+
+  labels = {
+    url = "http://www.acme.org"
+  }
+}
+
+data "google_monitoring_notification_channel" "my" {
+  display_name = google_monitoring_notification_channel.my.display_name
+}
+`, displayName)
+}
+
+func testAccDataSourceGoogleMonitoringNotificationChannel_byType(displayName string) string {
+	return fmt.Sprintf(`
+resource "google_monitoring_notification_channel" "my" {
+  display_name = "%s"
+  type         = "email"
```

Comment:
This type being reused causes a failure (below) when running in parallel, which is typical for our CI. I think we need a unique type here, that isn't reused in `testAccDataSourceGoogleMonitoringNotificationChannel_byDisplayNameAndType`. I'm not too familiar with this service- is there a 3rd we could use?

```
------- Stdout: -------
=== RUN   TestAccDataSourceGoogleMonitoringNotificationChannel_byType
--- FAIL: TestAccDataSourceGoogleMonitoringNotificationChannel_byType (2.85s)
    testing.go:635: Step 0 error: errors during apply:
        
        Error: More than one matching NotificationChannel found using filter=type="email"
        
          on /opt/teamcity-agent/temp/buildTmp/tf-test794806520/main.tf line 11:
          (source code not available)
        
        
    testing.go:696: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during refresh: More than one matching NotificationChannel found using filter=type="email"
        
        State: <nil>
FAIL
```

Reply:
yes that's possible, changed `testAccDataSourceGoogleMonitoringNotificationChannel_byType` to use `sms`